core/lib/navigation-error.js

```diff
@@ -46,6 +54,8 @@ function getNetworkError(mainRecord) {
       return new LighthouseError(
         LighthouseError.errors.FAILED_DOCUMENT_REQUEST, {errorDetails: netErr});
     }
+  } else if (mainRecord.statusCode === 404 && context.ignoreStatusCode) {
+    context.warnings.push(str_(UIStrings.warning404));
   } else if (mainRecord.hasErrorStatusCode()) {
```

Comment:
We should generalize this to all error status codes (4xx/5xx). So I think it's better to use this `mainRecord.hasErrorStatusCode()` block instead of doing a `=== 404` check.

Reply:
does generalizing this to include all error status codes change our initial thoughts on the defaults? or do they feel good still?

- devtools - ignore status code
- node -  fail on status code
- cli - fail on status code
- etc

Reply:
I think the defaults should be the same as we discussed